common/datavalues/src/data_array_arithmetic.rs

```diff
@@ -51,7 +51,8 @@ impl DataArrayArithmetic {
                         scalar.to_array_with_size(1)?
                     }
                 };
-                let result = Self::array_scalar_arithmetic_op(op, &left_array, right_value)?;
+                let right_array = right_value.to_array_with_size(left_array.len())?;
```

Comment:
This only works in the case of `constant % array`.

what about `constant +/-/*  array`?